cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -0,0 +1,804 @@
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {token_t::ErrorBegin},
+                                                        {},
+                                                        {},
+                                                        {token_t::ErrorBegin}}};
+  pda_tlt[static_cast<StateT>(pda_state_t::PD_ERR)] = {{{}, {}, {}, {}, {}, {}, {}, {}, {}, {},
+                                                        {}, {}, {}, {}, {}, {}, {}, {}, {}, {},
+                                                        {}, {}, {}, {}, {}, {}, {}, {}, {}, {}}};
+  return pda_tlt;
+}
+
+}  // namespace tokenizer_pda
+
+/**
+ * @brief Function object used to filter for brackets and braces that represent push and pop
+ * operations
+ */
+struct JSONToStackOp {
+  template <typename StackSymbolT>
+  constexpr CUDF_HOST_DEVICE fst::stack_op_type operator()(StackSymbolT const& stack_symbol) const
+  {
+    return (stack_symbol == '{' || stack_symbol == '[')   ? fst::stack_op_type::PUSH
+           : (stack_symbol == '}' || stack_symbol == ']') ? fst::stack_op_type::POP
+                                                          : fst::stack_op_type::READ;
+  }
+};
+
+namespace detail {
+
+void get_stack_context(device_span<SymbolT const> json_in,
+                       SymbolT* d_top_of_stack,
+                       rmm::cuda_stream_view stream)
+{
+  constexpr std::size_t single_item = 1;
+
+  // Symbol that will represent empty-stack (i.e., that we're at the DOM root)
```

Comment:
DOM root? Sounds like HTML/XML parsing and not JSON parsing. Maybe another choice of word is better.

Reply:
Agree 🙂   Fixed 👍